refactor!: remove compute plan status

thibowk opened the pull request:
## Description

<!-- Please reference issue if any. -->

<!-- Please include a summary of your changes. -->

Remove the compute plan status.

This will remove massive contention on the orchestrator and yet retain core functionality for the end user. Compute plan status will become a user-only concept computed by the backend: the existing duplicated logic in the orchestrator will be removed.

## Companion PR

- https://github.com/Substra/substra-backend/pull/465

## Changes

* Remove the `status` field in the compute plan protobuf schema 
* Add the `failure_date` field in the compute plan protobuf schema
* Add the `IsPlanRunning` RPC

## How has this been tested?

<!-- Please describe the tests that you ran to verify your changes.  -->

* [susbtra-tests](https://github.com/owkin/connect-ci/actions/runs/3096558302/jobs/5012250664)
* e2e orchestrator tests have been updated and are passing
* orchestrator unit tests have been updated and are passing

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [x] documentation was updated

thibowk commented:
/e2e

Owlfred commented:
End to end tests: :hourglass_flowing_sand: **STARTING**

Owlfred commented:
Something happened while processing your message: Unexpected inputs provided: ["testsBenchmark"]

thibowk commented:
/e2e --help

Owlfred commented:
```
Usage: /e2e [options] [help]

/e2e may appear anywhere as long as it is on its own line

Options:
  --refs <value>                                         Extra refs (branch or tag) with format REPO=GIT_REF,REPO=GIT_REF.
  Supported repositories: substra-hlf-k8s, orchestrator, substra-backend, substra-frontend, substra-tools, substrafl, substra, substra-tests, connect-ci.
  Example: /e2e --refs substra-backend=some_branch,orchestrator=some_tag (default: {})
  --tests-to-run, --tests <tests-to-run>                 Comma-separated list of tests to run. Valid options: sdk,substrafl,frontend or NONE. (default: "sdk")
  --benchmarks-to-run, --benchmarks <benchmarks-to-run>  Comma-separated list of workflows tests to run. Valid options: mnist, camelyon or NONE. (default: "NONE")
  --orchestrator-mode, --mode <orchestrator-mode>        Comma-separated list of orchestrator modes to run tests for. Valid options: standalone,distributed (default: "standalone,distributed")
  -h, --help                                             display help for command
```

Owlfred commented:
End to end tests: :hourglass_flowing_sand: **STARTING**

thibowk commented:
/e2e --refs substra-backend=remove-compute-plan-status

Owlfred commented:
End to end tests: :hourglass_flowing_sand: **STARTING**

thibowk commented:
/e2e --refs substra-backend=remove-compute-plan-status

Owlfred commented:
End to end tests: :hourglass_flowing_sand: **STARTING**

mblottiere commented on chaincode/computeplan/contract.go:
```diff
@@ -185,3 +185,33 @@ func (s *SmartContract) UpdatePlan(ctx ledger.TransactionContext, wrapper *commu
 	return nil
 }
+
+func (s *SmartContract) IsComputePlanRunning(ctx ledger.TransactionContext, wrapper *communication.Wrapper) (*communication.Wrapper, error) {
```
Please also [flag this method as read-only](https://github.com/Substra/orchestrator/blob/60ac7190810092bbc80ef481a2e14256e8eb0b58/server/common/evaluation.go#L22)

thibowk replied:
Done in f81cd4c799e47d1cc320e6c150b37e38859d39fe

thibowk commented on lib/errors/errors.go:
```diff
@@ -177,6 +180,11 @@ func NewMissingTaskOutput(taskKey, identifier string) *OrcError {
 }
 
+// NewTerminatedComputePlan returns an ErrTerminatedComputePlan kind of OrcError with given message
+func NewTerminatedComputePlan(msg string) *OrcError {
```
Addressed in 6939f87f472f4890c85190faf5aa86030592d0d3

thibowk commented:
/e2e --refs substra-backend=remove-compute-plan-status

Owlfred commented:
End to end tests: :hourglass_flowing_sand: **STARTING**

thibowk commented:
Thanks for your review @mblottiere.

I addressed your comments. Moreover, I renamed the method from `IsComputePlanRunning` to `IsPlanRunning` to be consistent with the other gRPC methods.

mblottiere commented on lib/service/computeplan.go:
```diff
@@ -15,8 +15,10 @@ type ComputePlanAPI interface {
 	QueryPlans(p *common.Pagination, filter *asset.PlanQueryFilter) ([]*asset.ComputePlan, common.PaginationToken, error)
 	ApplyPlanAction(key string, action asset.ComputePlanAction, requester string) error
 	UpdatePlan(computePlan *asset.UpdateComputePlanParam, requester string) error
+	FailPlan(plan *asset.ComputePlan) error
```
Accepting a CP means the caller has to fetch the plan beforehand: is there a rationale for accepting the plan rather than its key?
Also, since this should be called from task update, there is no need to make to method public (otherwise we would have to check that the requester has the right to fail the CP).
```suggestion
	failPlan(planKey string) error
```

thibowk replied:
Good points, thanks for raising them! There was no particular reason for doing that. I implemented the suggested changes in 6a5a605dd77062fa0c7fe1311e69bbb3e708c789.

mblottiere reviewed (commented):
Small question on the CP service API, otherwise LGTM

mblottiere reviewed (approved):
Thanks 👍 this should have a significant impact on contention
